main/guides/getting-started/contract-rpc.md

```diff
@@ -47,6 +47,24 @@ message MsgWalletSpendAction {
 
 ## Querying VStorage
 
+::: tip VStorage as Contract State Query Mechanism
+Unlike Ethereum and other blockchains that provide [readonly or view query APIs](https://docs.soliditylang.org/en/latest/contracts.html#view-functions) directly in the smart contracts,
+the Agoric chain does not offer such direct readonly APIs for querying contract state. Instead, the Agoric
+platform achieves this functionality through a powerful feature called VStorage.
+
+Smart contracts on the Agoric chain can write data to VStorage, which acts as a publicly accessible
+storage layer. This data is available for anyone to query, similar to the functionality of a getter
+API on other blockchains. It is completely cost-free, meaning developers and users can fetch data
+from VStorage without incurring any fees.
```

Comment:
While we don't have a vstorage-specific cost-recovery mechanism in place, there is a cost to the validators and such to store the data. For mainnet 2, we manually verify that the cost of storage used by contracts is paid for somehow. For  example, in Inter Protocol, there's a charge to open a vault and to keep it open.

As to fetching the data, operating an RPC server also has a cost. It's typically not born directly by developers or users, but there's no particular guarantee of availability. Rate limiting is typically used to keep the cost manageable, for example.

It's worth saying that fetching data does not incur the cost of a blockchain transaction. But "completely cost-free" over-states the case.